routes/user.js

```diff
@@ -52,58 +38,61 @@ async function userLogin(req, res){
     const userName = req.body.userName; 
     const password = req.body.password;
     const email = req.body.email; 
-    signInWithEmailAndPassword(auth, email, password)
-    .then((userCridential) => {
-        //Signed in
-        const user = userCridential.user;
-        res.send(user);
-    })
-    .catch((error) => {
-        const errorCode = error.code; 
-        const errorMessage = error.message; 
-        console.log("Error Code: ", errorCode);
-        console.log("Error Message: ", errorMessage);
-        res.send(errorMessage);
-      });
+    try {
+        const firebaseRep = await signInWithEmailAndPassword(auth, email, password);
+        const response = {
+            uid: firebaseRep.user.uid,
+            accessToken: firebaseRep.user.accessToken,
+            expirationTime: firebaseRep.user.stsTokenManager.expirationTime
+        };
+        res.send(response);
+    } catch (error) {
+        res.status(400)
+        res.send(error.message);
+    }
 }
 
 async function userSignup(req, res){
     const userName = req.body.userName;
     const password = req.body.password;
     const email = req.body.email;
 
-    //Creating a user with email and password
-    createUserWithEmailAndPassword(auth, email, password)
-    .then((userCridential) => {
-        //Signed in 
-        signUpUser(userCridential.user, userName)
+    try {
+        const firebaseRep = await createUserWithEmailAndPassword(auth, email, password);
         const response = {
-            accessToken: userCridential.user.accessToken,
-            expirationTime: userCridential.user.stsTokenManager.expirationTime
+            accessToken: firebaseRep.user.accessToken,
+            expirationTime: firebaseRep.user.stsTokenManager.expirationTime
         };
+        signUpUser(firebaseRep.user, userName)
         res.send(response);
-    })
-    .catch((error) => {
-        const errorCode = error.code; 
-        const errorMessage = error.message; 
-        console.log("Error Code: ", errorCode);
-        console.log("Error Message: ", errorMessage);
-        res.send(errorMessage);
-    })
+    } catch (error) {
+        res.status(400);
+        res.send(error.message);
+    }
 }
 
 async function addFavorite(req, res){
-    const favorites = req.body.favorites;
-    const userId = req.params.id;
-    const userFavorites = {
-        userId: userId,
-        favorites: favorites
+    const uid = req.params.id;
+    const movie = {
+        movieId: req.body.movieId,
+        imageUrl: req.body.imageUrl
     };
-    res.send(userFavorites);
+    try{
+        await addUserFavorite(uid,movie);
+        const response = {
+            uid: uid,
+            movieId: movie.movieId,
+            status: 'added'
```

Comment:
It's a little weird having a status when that's literally the purpose of the status code. You could use status 201 which means "created". But it's not important